### PR TITLE
feat: [CDS-80654]: Add sort dropdown in trigger explorer page for artifacts

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
#### Screenshots

https://github.com/harness/harness-core-ui/assets/98508399/24f0b56e-86be-4c85-9e0f-9064312ad00b

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/72-triggers/pages/triggers/views/TriggerExplorer.tsx [hunk 1]
```diff
@@ -123,6 +127,14 @@ const TriggerExplorer: React.FC = (): React.ReactElement => {
   const [showHelpPanel, setShowHelpPanel] = React.useState<boolean>(true)
   const [selectedArtifactTriggerTypeOption, setSelectedArtifactTriggerTypeOption] = React.useState<SelectOption>()
   const { accountId } = useParams<AccountPathProps>()
+  const { preference: sortPreference = SortMethod.Newest, setPreference: setSortPreference } =
+    usePreferenceStore<SortMethod>(PreferenceScope.USER, `sort-TriggersExplorerPage`)
+
+  const sortByTriggerIdentifier: SelectOption[] = [
+    { label: 'Name (A->Z, 0->9)', value: 'triggerIdentifier,ASC' },
```
Comment (pankajsingh-harness) [inline src/modules/72-triggers/pages/triggers/views/TriggerExplorer.tsx:134]: Is this shorting based on name or identifier in BE? Please confirm once with BE.

Also, the label should come from yaml. 

Can we not use `src/modules/10-common/components/SortOption/SortOption.tsx` with some changes here?

Comment (pankajsingh-harness) [reply src/modules/72-triggers/pages/triggers/views/TriggerExplorer.tsx:134]: Refer to sort functionality at other places.

Comment (sonaligoyal-harness) [reply src/modules/72-triggers/pages/triggers/views/TriggerExplorer.tsx:134]: Yes, i already did and confirmed with BE.
Here, in BE they only have identifier in response due to which we have to send like this.

**Review by pankajsingh-harness — COMMENTED** [review]
@sonaligoyal-harness can we ask BE to have the same naming for the filter value as other places? We are using `name` as the identifier for the pipeline filter.

**Comment by pankajsingh-harness** [comment]
Refer to the sort functionality for the Pipeline page

**Review by pankajsingh-harness — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Pull request merged**